apps/lib/s_socket.c

```diff
@@ -173,7 +173,7 @@ int init_client(int *sock, const char *host, const char *port,
         }
 
         /* Save the address */
-        if (tfo || !doconn)
+        if ((tfo || !doconn) && (ba_ret != NULL))
```

Comment:
`ba_ret` can never be NULL. This is a pointless check.

Reply:
I don't see a `NULL` check for `ba_ret` in this function, and it's not `static`, and `ba_ret` isn't documented as "must not be NULL" so while yes, currently it's only called in one place and `ba_ret` isn't `NULL` in that case, but that doesn't mean this is a pointless check

Reply:
Then it should be done like:
```suggestion
        if (tfo || !doconn) {
            if (ba_ret != NULL) {
                BIO_printf(bio_err, "Internal error\n");
                BIO_closesocket(*sock);
                goto out;
            }
            ...
```